zcash_primitives/src/transaction/builder.rs

```diff
@@ -334,8 +335,11 @@ impl<'a, P: consensus::Parameters, R: RngCore + CryptoRng> Builder<'a, P, R> {
         self.sapling_builder
             .add_spend(&mut self.rng, &extsk, note, merkle_path)?;
 
-        self.sapling_asks
-            .push(redjubjub::PrivateKey(extsk.expsk.ask));
+        // TODO: store a `redjubjub::SigningKey` inside `extsk`.
```

Comment:
I'm not sure that I agree with this TODO; it seems like it would be storing duplicate data. If the cost of recomputing the signing key is a problem then I guess that caching it is okay, but we should motivate such caching with a benchmark.

Reply:
It wouldn't be storing duplicate data; it would be replacing the `ask` currently stored there. This is equivalent to how the `orchard` key structs work.